storage/storage_image.go

```diff
@@ -336,21 +349,34 @@ func (s *storageImageDestination) Commit() error {
 	}
 	img, err := s.imageRef.transport.store.CreateImage(s.ID, nil, lastLayer, "", nil)
 	if err != nil {
-		logrus.Debugf("error creating image: %q", err)
-		return err
+		if err != storage.ErrDuplicateID {
+			logrus.Debugf("error creating image: %q", err)
+			return err
+		}
+		img, err = s.imageRef.transport.store.GetImage(s.ID)
```

Comment:
This ignores `err`, so in principle could crash if `img` is `nil`?

AFAICT `s.ID`, via `ParseStoreReference`… `newImageDestination`, can end up empty. What does `GetImage` do with that case? Or is the argument that `ErrDuplicateID` necessarily means that `GetImage` will succeed and return an image?